### PR TITLE
Add mujin_deletefile sample

### PR DESCRIPTION
**Pull request by cielavenir**

### CODE REVIEW

**Comment by ziyan** [comment]
UploadFile does not take the `/u/username/` path, I am not sure if we should do that for `DeleteFile`, it is not symmetric.

**Comment by cielavenir** [comment]
@ziyan I'm doing:

```
void ControllerClientImpl::UploadFileToController_UTF8(const std::string& filename, const std::string& desturi)
{
    boost::mutex::scoped_lock lock(_mutex);
    _UploadFileToController_UTF8(filename, _PrepareDestinationURI_UTF8(desturi, false));
}

void ControllerClientImpl::DeleteFileOnController_UTF8(const std::string& desturi)
{
    boost::mutex::scoped_lock lock(_mutex);
    _DeleteFileOnController(_PrepareDestinationURI_UTF8(desturi, false));
}
```

I suppose this is taking `/u/username` path, right?

**Comment by ziyan** [comment]
Is this taking `mujin:/dest`?

File: src/controllerclientimpl.cpp [hunk 1]
```diff
@@ -1787,21 +1785,16 @@ void ControllerClientImpl::_UploadDataToController(const std::vector<unsigned ch
...
-        else {
-            throw MUJIN_EXCEPTION_FORMAT("delete failed with HTTP status %s", desturi%http_code, MEC_HTTPServer);
-        }
+    MUJIN_LOG_DEBUG(str(boost::format("delete %s")%desturi))
+
+    // the dest filename of the upload is determined by stripping the leading _basewebdavuri
+    if( desturi.size() < _basewebdavuri.size() || desturi.substr(0,_basewebdavuri.size()) != _basewebdavuri ) {
+        throw MUJIN_EXCEPTION_FORMAT("trying to upload a file outside of the webdav endpoint is not allowed: %s", desturi, MEC_HTTPServer);
     }
+    std::string filename = desturi.substr(_basewebdavuri.size());
```
Comment (ziyan) [inline src/controllerclientimpl.cpp]: Is `_basewebdavuri` here `/u/username`?

If so, it feels weird that we are converting from `mujin:/abc.mujin.dae` first to `/u/username/abc.mujin.dae` then to `abc.mujin.dae'

**Comment by cielavenir** [comment]
@ziyan Yes `mujin:/dest.mujin.dae`.

```
controllerclient->UploadFileToController_UTF8(filename, std::string("mujin:/")+basename);

controllerclient->DeleteFileOnController_UTF8(std::string("mujin:/")+remotefilename);
```

File: src/controllerclientimpl.cpp (hunk 1 above)
Comment (cielavenir) [reply src/controllerclientimpl.cpp]: > Is _basewebdavuri here /u/username?

Yes.

_PrepareDestinationURI_UTF8 needs to be rewritten. However the customer needs DeleteFileOnController_UTF8 API as soon as possible...

Comment (cielavenir) [reply src/controllerclientimpl.cpp]: ...Actually, what should happen if I specify something not-including "mujin:/" as desturi?

Comment (ziyan) [reply src/controllerclientimpl.cpp]: If the customer needs the API as soon as possible, I am okay with current changes as long as it works.

**Comment by kanbouchou** [comment]
Looks good to me.

**Comment by rdiankov** [comment]
thanks!

**Pull request merged**